Fortnox.NET.Tests/Fortnox.NET.Tests/WebSocket/FortnoxWebSocketClientTest.cs

```diff
@@ -0,0 +1,262 @@
+﻿using FortnoxNET.Communication;
+using FortnoxNET.Communication.Order;
+using FortnoxNET.Models.Article;
+using FortnoxNET.Models.Order;
+using FortnoxNET.Services;
+using FortnoxNET.WebSockets;
+using FortnoxNET.WebSockets.Models;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using Newtonsoft.Json;
+using System;
+using System.Net.WebSockets;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace FortnoxNET.Tests.WebSocket
+{
+    [TestClass]
+    public class FortnoxWebSocketClientTest : TestBase
+    {
+        private const int DEFAULT_BUFFER_SIZE = 2048;
+
+        [TestMethod]
+        public async Task CanListenUsingEnumerable()
+        {
+            var client = new FortnoxWebSocketClient(this.connectionSettings.AccessToken, this.connectionSettings.ClientSecret)
+                .AddTopic(WebSocketTopic.Articles);
+
+            var listeningTask = new Task(async () =>
+            {
+                try
+                {
+                    (await client.Connect()).Listen(async (socket) =>
+                    {
+                        foreach (var response in client.GetNextEvent(socket))
```

Comment:
awesome! 🥳

Reply:
Personally, I think this is the API we should promote, as we've spoken about previously @brokenprogrammer. I see that you've done so in the docs, which is awesome!

Reply:
FYI @BaconFinger